sammo.py

```diff
@@ -3,18 +3,48 @@
 __contact__ = "info@hytech-imaging.fr"
 __copyright__ = "Copyright (c) 2021 Hytech Imaging"
 
-from .src.gui.session import SammoActionSession
+from .src.gui.session import SammoActionSession, ParentOfSammoActionSession
+from .src.gui.onOffEffort import SammoActionOnOffEffort, ParentOfSammoActionOnOffEffort
+from .src.core.session import SammoSession
 
 
-class Sammo:
+class Sammo(ParentOfSammoActionSession, ParentOfSammoActionOnOffEffort):
     def __init__(self, iface):
         self.iface = iface
-        self.toolBar = self.iface.addToolBar("Sammo ToolBar")
-        self.actionSession = SammoActionSession(self.iface.mainWindow(), self.toolBar)
+        self._toolBar = self.iface.addToolBar("Sammo ToolBar")
+        self._session = SammoSession()
+        self.actionSession = SammoActionSession(self)
+        self.actionOnOffSession = SammoActionOnOffEffort(self)
 
     def initGui(self):
         self.actionSession.initGui()
+        self.actionOnOffSession.initGui()
 
     def unload(self):
         self.actionSession.unload()
-        del self.toolBar
+        self.actionOnOffSession.unload()
+        del self._toolBar
+
+    def onCreateSession(self, workingDirectory : str):
+        self._session.onCreateSession(workingDirectory)
+        self.actionOnOffSession.onCreateSession()
+
+    def OnStartEffort(self):
+        [feat, table] = self._session.getReadyToAddNewFeatureToEnvironmentTable()
+        if self.iface.openFeatureForm(table, feat):
+            self._session.addNewFeatureToEnvironmentTable(feat)
```

Comment:
I think that this part should be hidden in `SammoActionOnOffEffort`.

Reply:
Are you sure ? 
I think it is much simple / safe this way because we don't have to provide "session" and "iface" to SammoActionOnOffEffort.
Besides, I think it is not the responsability of this GUI class to know what an effort is.

Reply:
Actually the `iface.openFeatureForm` is a GUI action (because a form is opened) so I think it makes sense to move this call in `SammoActionOnOffEffort`. 

> Besides, I think it is not the responsability of this GUI class to know what an effort is.

Indeed, you're right. If we move the `iface.openFeatureForm` call in `SammoActionOnOffEffort`, we could emit a signal when the form is closed. Then `Sammo` catch the signal and add the feature in the table. What do you think?

Reply:
yes, I also noticed that openFeatureForm is a GUI action :)
but it is so disconnected from SammoActionOnOffEffort that it seemed to me simpler to put it there, than implementing in SammoActionOnOffEffort the call to "_session.getReadyToAddNewFeatureToEnvironmentTable()" we need before and the call to "_session.addNewFeatureToEnvironmentTable(feat)" we need after the call to openFeatureForm.

I will figure something out.